File: perl/dataprocess/cal_decadal.pl
#!/usr/bin/perl -w

# Description: Calculate decadal from yearly data of PCOM
#
#       Usage: ./xxx.pl
#
#      Author: OU Yuyuan <ouyuyuan@lasg.iap.ac.cn>
#     Created: 2013-09-16 21:32:38 CST
# Last Change: 2013-09-17 18:44:37 CST

use strict;
use 5.010;

my $decade = 1;
my $nyear = 100;
my $ndecade = int($nyear/10);
my $src_dir = "yearly";
my $des_dir = "decadal";

while ($decade <= $ndecade) {
   my $year = 1;
   my $total_year;
   my $filename;
   my $cmd = "cdo ensmean ";

   while ($year <= 10) {
      $total_year = ($decade-1)*10 + $year;
      my $filename = sprintf("%s%04s%s","$src_dir/year_",$total_year,".nc");

      $cmd .= $filename . " ";
      $year += 1;
   }
   $cmd .= sprintf("%s%02s%s","$des_dir/decade_",$decade,".nc");

   say $cmd;
   system($cmd);

   $decade += 1;
}

File: calculate/pcom/cal_decadal.py
#!/usr/bin/env python

# Description: cal. climate mean of PCOM monthly output data
#
#      Author: OU Yuyuan <ouyuyuan@lasg.iap.ac.cn>
#     Created: 2013-12-12 10:02:20 BJT
# Last Change: 2014-06-11 06:42:03 BJT

import os
import sys

dc_sta = 1
dc_end = 50

for decade in range(dc_sta, dc_end + 1):
  yr_sta = (decade - 1) * 10 + 1
  yr_end = decade * 10
  filenames = ''
  for i in range(yr_sta, yr_end + 1):
    f = "year_%04d.nc " % i
    filenames += f
  cmd = 'cdo ensmean ' + filenames +  ' decade_%02d.nc' % decade
  print(cmd)
  os.system(cmd)
